src/persist-client/src/internal/compact.rs

```diff
@@ -305,16 +305,16 @@ where
         let compact_span = debug_span!("compact::consolidate");
         let res = tokio::time::timeout(
             timeout,
-            // Compaction is cpu intensive, so be polite and spawn it on the CPU heavy runtime.
-            cpu_heavy_runtime
+            // Compaction is cpu intensive, so be polite and spawn it on the isolated runtime.
```

Comment:
hmm I'm suddenly a little stressed that this intentionally spawns compaction in a normal, non-blocking task and we'll be co-located with it. maybe we should put an artificial await in compaction somehwere? i don't recall the full context, but it looks like the reason we had CpuHeavyRuntime in the first place was to avoid spawn_blocking for some shutdown-related reason

Reply:
hm do you have more context on the shutdown issue? I don't have a good sense of why one approach is preferable to the other

Reply:
i don't remember any context but it should be in the PR that added CpuHeavyRuntime, i'll pull that up. what's the other approach?

Reply:
https://github.com/MaterializeInc/materialize/pull/13955

Reply:
>hmm I'm suddenly a little stressed that this intentionally spawns compaction in a normal, non-blocking task and we'll be co-located with it. maybe we should put an artificial await in compaction somehwere?

We actually have an await now every time we add an update to the batch builder. The longest time without an await point is decoding an individual part into the heap, so up to 128MiB worth of decoding. I don't have a number for how long this could be, but offhand seems like it shouldn't be bad?

Reply:
hmm which await point? i _think_ that just calling `.await` is not sufficient if e.g. the `async` function you're awaiting doesn't itself have an await point. there's got to be a real future being polled down at the bottom of the compile async/await state machine somewhere

Reply:
oooh you're totally right. okay so then the max time without an await could be fetching / decoding a 128MiB part, followed by encoding a new 128MiB part. think it's worth throwing in a `yield_now` in there? maybe after decoding / encoding N updates?

Reply:
yeah that sounds like exactly what I was thinking

Reply:
added a fixed yield threshold. opted to not make it dynamic partially out of laziness, partially because I can't imagine what signal we'd see in our metrics to decide to change it... I suppose we could also start timing how long we go without yielding, but, the value there starts to feel pretty marginal